File: Perl/fizzbuzz.pl
#! /usr/bin/perl
# As we all know, there is only one proper language for FizzBuzz, and only one graceful
# way to accomplish it, the Perl 'for' loop. HA!
#
# now I am tempted to Perl-golf the heck out of this....

for (1..100) {
    my $fizz = ($_ % 3 == 0)? "Fizz" : "";
    my $buzz = ($_ % 2 == 0)? "Buzz" : "";
    ($fizz ne '' || $buzz ne '') ? print "$fizz$buzz\n" : print "$_\n";
}

File: Ruby/fizzbuzz.rb
(1..100).each do |number|
  multiple_of_three = (number % 3) == 0
  multiple_of_five  = (number % 5) == 0
  multiple_of_both  = multiple_of_three && multiple_of_five
  if multiple_of_both
    puts "FizzBuzz"
  elsif multiple_of_three
    puts "Fizz"
  elsif multiple_of_five
    puts "Buzz"
  else
    puts number
  end
end

